packages/aft/lib/src/commands/generate/generate_amplify_swift_command.dart

```diff
@@ -0,0 +1,288 @@
+
+  @override
+  String get name => 'amplify-swift';
+
+  @override
+  bool get hidden => true;
+
+  /// The branch of Amplify Swift to target.
+  ///
+  /// If not provided, defaults to `release`.
+  late final branchTarget = argResults!['branch'] as String? ?? 'release';
+
+  /// The lib/-relative output path for the generated plugins.
+  late final outputPath = argResults!['output'] as String? ??
+      'packages/amplify_datastore/ios/internal';
+
+  /// Whether to check the diff of the generated files.
+  /// If not provided, defaults to `false`.
+  late final isDiff = argResults!['diff'] as bool? ?? false;
+
+  /// Cache of repos by git ref.
+  final _repoCache = <String, Directory>{};
+  final _cloneMemo = AsyncMemoizer<Directory>();
+
+  final _amplifySwiftPlugins = [
+    const PluginConfig(
+      name: 'AWSDataStorePlugin',
+      remotePathToSource: 'AmplifyPlugins/DataStore/Sources/AWSDataStorePlugin',
+    ),
+    const PluginConfig(
+      name: 'AWSPluginsCore',
+      remotePathToSource: 'AmplifyPlugins/Core/AWSPluginsCore',
+    ),
+    const PluginConfig(
+      name: 'Amplify',
+      remotePathToSource: 'Amplify',
+    ),
+  ];
+
+  /// Downloads Amplify Swift from GitHub into a temporary directory.
+  Future<Directory> _downloadRepository() => _cloneMemo.runOnce(() async {
+        final cloneDir =
+            await Directory.systemTemp.createTemp('amplify_swift_');
+        logger
+          ..info('Downloading Amplify Swift...')
+          ..verbose('Cloning repo to ${cloneDir.path}');
+        await runGit(
+          [
+            'clone',
+            // https://github.blog/2020-12-21-get-up-to-speed-with-partial-clone-and-shallow-clone/
+            '--filter=tree:0',
+            'https://github.com/aws-amplify/amplify-swift.git',
+            cloneDir.path,
+          ],
+          echoOutput: verbose,
+        );
+        logger.info('Successfully cloned Amplify Swift Repo');
+        return cloneDir;
+      });
+
+  /// Checks out [ref] in [pluginDir].
+  Future<Directory> _checkoutRepositoryRef(
+    Directory pluginDir,
+    String ref,
+  ) async {
+    logger
+      ..info('Checking out target branch: $ref')
+      ..verbose('Creating git work tree in $pluginDir');
+    final worktreeDir =
+        await Directory.systemTemp.createTemp('amplify_swift_worktree_');
+    try {
+      await runGit(
+        ['worktree', 'add', worktreeDir.path, ref],
+        processWorkingDir: pluginDir.path,
+      );
+    } on Exception catch (e) {
+      if (e.toString().contains('already checked out')) {
+        return pluginDir;
+      }
+      rethrow;
+    }
+    return worktreeDir;
+  }
+
+  /// Sets up the Amplify Swift repo for use later
+  Future<void> _setupRepo() async {
+    if (_repoCache[branchTarget] != null) {
+      return;
+    }
+    Directory repoDir;
+    repoDir = await _downloadRepository();
+    repoDir = await _checkoutRepositoryRef(repoDir, branchTarget);
+
+    _repoCache[branchTarget] = repoDir;
```

Comment:
Suggestion: You can make `repoDir` final if you refactor this a bit

```suggestion
    final repoDir = await _checkoutRepositoryRef(await _downloadRepository(), branchTarget);

    _repoCache[branchTarget] = repoDir;
```

Reply:
Alternatively, something like this would work as well.

```suggestion
    final repoDir = await _downloadRepository();
    final repoRef = await _checkoutRepositoryRef(repoDir, branchTarget);

    _repoCache[branchTarget] = repoRef;
```

Reply:
Good suggestion, took the later.